### ISSUE TITLE
I had some problems training stpls3d .

### ISSUE THREAD

**Issue by buggogogo** [JonasSchult/Mask3D#64]
Hello Author！
 Your research is excellent and I benefit a lot from it.I configured the training environment in accordance with "read.md", installed the required packages in accordance with "requirements.txt", put stpls3d in /data/raw/stpls3d folder and divided train, validation and test（15:5:5）.

After I run stpls3d_preprocessing.py in Pycharm , I tried to  run main_instance_segmentation.py in Pycharm Terminal with following commands (copy from stpls3d_val.sh):

 python main_instance_segmentation.py general.experiment_name="validation_query_${CURR_QUERY}_topk_${CURR_TOPK}_dbscan_${CURR_DBSCAN}_size_${CURR_SIZE}" general.project_name="stpls3d_eval" data/datasets=stpls3d general.num_targets=15 data.num_labels=15 data.voxel_size=0.333 data.num_workers=10 data.cache_data=true data.cropping_v1=false general.reps_per_epoch=100 model.num_queries=${CURR_QUERY} general.on_crops=true model.config.backbone._target_=models.Res16UNet18B general.train_mode=false general.checkpoint="checkpoints/stpls3d/stpls3d_val.ckpt" data.crop_length=${CURR_SIZE} general.eval_inner_core=50.0 general.topk_per_image=${CURR_TOPK} general.use_dbscan=true general.dbscan_eps=${CURR_DBSCAN}

The program finished running in a few seconds. I didn't see any error messages, but I didn't find any save results.
The output in pycharm's terminal is as follows:

(mask3d) zy@zy-cau:~/3D_Instance_Segmentation/Mask3D$ export OMP_NUM_THREADS=3
(mask3d) zy@zy-cau:~/3D_Instance_Segmentation/Mask3D$ CURR_DBSCAN=14.0
(mask3d) zy@zy-cau:~/3D_Instance_Segmentation/Mask3D$ CURR_TOPK=750
(mask3d) zy@zy-cau:~/3D_Instance_Segmentation/Mask3D$ CURR_QUERY=160
(mask3d) zy@zy-cau:~/3D_Instance_Segmentation/Mask3D$ CURR_SIZE=54
(mask3d) zy@zy-cau:~/3D_Instance_Segmentation/Mask3D$ python main_instance_segmentation.py \
> general.experiment_name="validation" \
> general.project_name="stpls3d" \
> data/datasets=stpls3d \
> general.num_targets=15 \
> data.num_labels=15 \
> data.voxel_size=0.333 \
> data.num_workers=10 \
> data.cache_data=true \
> data.cropping_v1=false \
> general.reps_per_epoch=100 \
> model.num_queries=${CURR_QUERY} \
> general.on_crops=true \
> model.config.backbone._target_=models.Res16UNet18B \
> data.crop_length=${CURR_SIZE} \
> general.eval_inner_core=50.0

wandb: Tracking run with wandb version 0.13.2
wandb: W&B syncing is set to `offline` in this directory.  
wandb: Run `wandb online` or set WANDB_MODE=online to enable cloud syncing.
/home/zy/anaconda3/envs/mask3d/lib/python3.10/site-packages/pytorch_lightning/utilities/seed.py:55: UserWarning: No seed found, seed set to 1183304741
  rank_zero_warn(f"No seed found, seed set to {seed}")
Global seed set to 1183304741
{'_target_': 'pytorch_lightning.loggers.WandbLogger', 'project': '${general.project_name}', 'name': '${general.experiment_name}', 'save_dir': '${general.save_dir}', 'entity': 'schult', 'resume': 'allow', 'id': '${general.experiment_name}'}
/home/zy/anaconda3/envs/mask3d/lib/python3.10/site-packages/pytorch_lightning/loggers/wandb.py:352: UserWarning: There is a wandb run already in progress and newly created instances of `WandbLogger` will reuse this run. If this is not desired, call `wandb.finish()` before instantiating `WandbLogger`.
  rank_zero_warn(

/home/zy/anaconda3/envs/mask3d/lib/python3.10/site-packages/pytorch_lightning/trainer/connectors/accelerator_connector.py:446: LightningDeprecationWarning: Setting `Trainer(gpus=1)` is deprecated in v1.7 and will be removed in v2.0. Please use `Trainer(accelerator='gpu', devices=1)` instead.
  rank_zero_deprecation(
/home/zy/anaconda3/envs/mask3d/lib/python3.10/site-packages/pytorch_lightning/trainer/connectors/callback_connector.py:57: LightningDeprecationWarning: Setting `Trainer(weights_save_path=)` has been deprecated in v1.6 and will be removed in v1.8. Please pass ``dirpath`` directly to the `ModelCheckpoint` callback
  rank_zero_deprecation(
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
generate data/processed/stpls3d/train_database.yaml first
wandb: Waiting for W&B process to finish... (success).
wandb:                                                                                
wandb: You can sync this run to the cloud by running:
wandb: wandb sync /home/zy/3D_Instance_Segmentation/Mask3D/wandb/offline-run-20230426_200434-1lt52f5z
wandb: Find logs at: ./wandb/offline-run-20230426_200434-1lt52f5z/logs

Thank you for your help and I am looking forward to your reply.

**Comment by JonasSchult** [comment]
Hi!

Thanks for your interest in our work. :)

This is the "key" line:
> generate data/processed/stpls3d/train_database.yaml first

You need to process the dataset first. See [here](https://github.com/JonasSchult/Mask3D#stpls3d).

Best,
Jonas

**Comment by monarchdemon** [comment]
I have a similar issue above, have you solved it?

**Comment by JonasSchult** [comment]
Hi! :)

Thanks for your interest in our work!

we have just released a [fix](https://github.com/JonasSchult/Mask3D/commit/9f25bf357c97ed7b80402a6233eee8d940c64aeb) which most likely will solve your issue. Just pull the latest version of the project and recreate your virtual python environment following the instructions [here](https://github.com/JonasSchult/Mask3D#dependencies-memo).

Previously, the issue occurred because we did not lock the packages in the Python environment to specific versions. We now made sure that the packages in the environment remain the same.

I will close the issue now. Please feel free to reopen it again if you continue to experience issues.

Best,
Jonas
